hs_discover/views.py

```diff
@@ -1,57 +1,226 @@
+import datetime
```

Comment:
the main endpoint for new discover, combining geo features, haystack interactions, and API logic for building the search page